sympy/combinatorics/free_group.py

```diff
@@ -806,6 +831,58 @@ def subword(self, from_i, to_j):
             array_form = letter_form_to_array_form(letter_form, group)
             return group.dtype(array_form)
 
+    def is_independent(self, word):
```

Comment:
Perhaps this is the line from Havas. From second paragraph beginning (`elimination_technique_1`)

> provided the relator is independent of previously discovered but not yet eliminated redundant generators). 

I am not fully sure that I even understood its meaning correctly.